### PR TITLE
AsRawHandle IntoRawHandle and JoinHandleExt for JoinHandle

### PR DESCRIPTION
**Pull request by retep998**
Allows a `HANDLE` to be extracted from a `JoinHandle` on Windows.
Allows a `pthread_t` to be extracted from a `JoinHandle` everywhere else.

Because https://github.com/rust-lang/rust/pull/29461 was closed.

r? @alexcrichton

### CODE REVIEW

File: src/libstd/os/openbsd/raw.rs [hunk 1]
```diff
@@ -25,6 +25,8 @@ use os::unix::raw::{uid_t, gid_t};
 #[stable(feature = "raw_ext", since = "1.1.0")] pub type off_t = i64;
 #[stable(feature = "raw_ext", since = "1.1.0")] pub type time_t = i64;
 
+#[unstable(feature = "pthread_t", issue = "0")] pub type pthread_t = usize;
```
Comment (alexcrichton) [inline src/libstd/os/openbsd/raw.rs]: Could you connect all these issues to the same tracking issue as below? (#29791)

**Comment by alexcrichton** [comment]
Thanks! Just one nit and otherwise r=me

**Comment by retep998** [comment]
Addressed nit.

**Comment by alexcrichton** [comment]
@bors: r+ 9376f707524f291021a5621ae7dccf20db996027

Thanks!

File: src/libstd/sys/windows/ext/thread.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Extensions to `std::thread` for Windows.
+
```
Comment (retep998) [inline src/libstd/sys/windows/ext/thread.rs]: Oh dang, I need a stability attribute here. Which feature do I use?

Comment (alexcrichton) [inline src/libstd/sys/windows/ext/thread.rs]: It's fine to leave the module unstable (there's actually no need for it to be public), and just point it to the same pthread_t issue.

**Comment by retep998** [comment]
Added that missing stability attribute.

**Pull request merged**